File: common-web/src/main/webapp/js/service/imageCaseInfoService.js
//服务层
app.service('imageCaseInfoService', function($http) {

	// 读取列表数据绑定到表单中
	this.findAll = function() {
		return $http.get('../imageCaseInfo/findAll.do');
	}
	// 分页
	this.findPage = function(page, rows) {
		return $http.get('../imageCaseInfo/findPage.do?page=' + page + '&rows='
				+ rows);
	}
	// 查询实体
	this.findOne = function(id) {
		return $http.get('../imageCaseInfo/findOne.do?id=' + id);
	}
	// 增加
	this.add = function(entity) {
		return $http.post('../imageCaseInfo/add.do', entity);
	}
	// 修改
	this.update = function(entity) {
		return $http.post('../imageCaseInfo/update.do', entity);
	}
	// 删除
	this.dele = function(ids) {
		return $http.get('../imageCaseInfo/delete.do?ids=' + ids);
	}
	// 搜索
	this.search = function(page, rows, searchEntity) {
		return $http.post('../imageCaseInfo/search.do?page=' + page + "&rows="
				+ rows, searchEntity);
	}

	// 通过报案号和图片名称搜索
	this.searchByCaseIdAndImgName = function(caseId, imgName) {
		return $http.get('../imageCaseInfo/searchByCaseIdAndImgName.do?caseId='
				+ caseId + "&imgName=" + imgName);
	}

	// 获取案件最新状态
	this.lastedStatus = function(caseId) {
		return $http.get('../imageCaseInfo/findLastedStatusByCaseId.do?caseId='
				+ caseId);
	}

	
	//reSend
	this.reSend = function(caseId) {
		return $http.get('../imageCaseInfo/reSend.do?caseId='
				+ caseId);
	}
});

File: common-web/src/main/java/com/pyg/manager/dao/ImageCaseInfoService.java
package com.pyg.manager.dao;
import java.util.List;


import com.pyg.manager.dao.pojo.TbImageCaseInfo;

import entity.PageResult;
/**
 * 服务层接口
 * 
 * @author Administrator
 *
 */
public interface ImageCaseInfoService {

	/**
	 * 返回全部列表
	 * 
	 * @return
	 */
	public List<TbImageCaseInfo> findAll();

	/**
	 * 返回分页列表
	 * 
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 增加
	 */
	public void add(TbImageCaseInfo imageCaseInfo);

	/**
	 * 修改
	 */
	public void update(TbImageCaseInfo imageCaseInfo);

	/**
	 * 根据ID获取实体
	 * 
	 * @param id
	 * @return
	 */
	public TbImageCaseInfo findOne(Long id);

	/**
	 * 批量删除
	 * 
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * 
	 * @param pageNum
	 *            当前页 码
	 * @param pageSize
	 *            每页记录数
	 * @return
	 */
	public PageResult findPage(TbImageCaseInfo imageCaseInfo, int pageNum,
                               int pageSize);

	public List<TbImageCaseInfo> findByCaseId(String caseId);

	/**
	 * 日期区间查询
	 * @param startDate
	 * @param endDate
	 * @return
	 */
	List<TbImageCaseInfo> findByDate(String curUser, String startDate, String endDate, String caseStatus);
}
